internal/server/server.go

```diff
@@ -482,6 +482,9 @@ func handleShutdown(ctx context.Context, rc requestContext) (interface{}, *apiEr
 }
 
 func (s *Server) requestShutdown(ctx context.Context) {
+	// Workaround for passing tests
+	s.rep.Metrics().RemoveAllGauges()
```

Comment:
i think that's a bit too early as during shutdown we may emit some gauges.

Also, can this RemoveAllGauges be done in the test itself and not in the server?

Reply:
The worse thing can happen is prom will scrape /metrics that moment and think that these metrics have been removed.
As putting in testing, I think is not clean call an internal function while doing e2e.
Ideally this should be called after the metrics listener has been closed. BTW is this getting closed with server shutdown? I am not that sure